### ISSUE TITLE
Potential mistake in iOS documentation regarding using globstar in Input Files list

### ISSUE THREAD

**Issue by wnagrodzki** [apollographql/apollo-ios#914]
**Issue**

Installation instruction listed on [this web page](https://www.apollographql.com/docs/ios/installation/) advises to place `$(SRCROOT)/YourTarget/**/*.graphql` into _Input Files_ to prevent unnecessary recompilations.

I discovered that `API.swift` is not regenerated if one of the _graphql_ files is modified.

**Diagnosis**

It looks like globstar is not respected. Tested with Xcode 11.2.1 (11B500).
After checking environment variables I see this is treated as one file (globstar is not expanded).
```
export SCRIPT_INPUT_FILE_0="/Users/wna/Developer/Trifork/Trifork - Supertrends/Supertrends-iOS/iOS/Supertrends/GraphQL/**/*.graphql"
```

**Solution**
I ended up generating `xcfilelist` and passing it as argument to _Input File Lists_.

I am not sure if this is Xcode regression, or a mistake in Apollo documentation. I did not find any information in Xcode help that globstar is supported for _Input Files_. Can anyone confirm if globstar is working?

**Comment by designatednerd** [comment]
Ergh, putting that into the input files should say to explicitly look for all the files. 

I saw there's a space in your path - `Trifork - Supertrends`. What happens if there's no space in the path? I know there's a lot of built-in stuff with Xcode that doesn't work like it should when you use spaces in your path.

**Comment by wnagrodzki** [comment]
I am able to reproduce the same behaviour without spaces. `env` lists as it was only one file.
```
SCRIPT_INPUT_FILE_0=/Users/wna/Developer/Trifork/Trifork-Supertrends/Supertrends-iOS/iOS/Supertrends/GraphQL/**/*.graphql
```

Do you see a different behaviour on your machine?

**Comment by designatednerd** [comment]
I'm gonna have to take a longer look at this later this week - but I definitely have generating an XCFileList as part of the Swift Codegen rewrite todos.

**Comment by francybiga** [comment]
I also noticed that globbing (as suggested by [Apollo docs](https://www.apollographql.com/docs/ios/installation/#prevent-unnecessary-recompilation) isn't working as expected (I'm using Xcode-12.3). I managed to obtain the desired result generating an `xcfilelist` and using it instead.

Also note that, as explained in this [WWDC video](https://developer.apple.com/videos/play/wwdc2018/408/), `xcfilelist` "cannot be modified or generated throughout your build process.", so it's not possible perform it in a Run Script build phase. 

I generate the `xcfilelist`as a Schema Pre-Action for the Build action. Remember to select your target in "Provide build settings from".

<img width="1280" alt="Screenshot 2021-01-20 at 16 47 02" src="https://user-images.githubusercontent.com/244899/105199816-7d688080-5b3f-11eb-8497-14d32ecd5f67.png">

The script content is something like this:

```bash
#!/bin/sh

FILE_LIST_PATH="${SRCROOT}/Prima/Network/GraphQL/apollo_codegen_input.xcfilelist"
TMP_PATH="/tmp/apollo_codegen_input.xcfilelist"

# We change current dir since we want to list relative paths
cd ${SRCROOT} 

# Write list of graphql files to a tmp file, adding the $(SRCROOT) prefix to the file relative path
find . -name "*.graphql" | sed -e 's/^\./$(SRCROOT)/' > ${TMP_PATH}

# Only ovewrite file list with new content if content differs
# If we simply output to the file list each time, Xcode considers it as a modification,
# thus will run the Apollo Codegen script even if no file has changed 

if ! diff -q "${TMP_PATH}" "${FILE_LIST_PATH}" 
then    
    mv "${TMP_PATH}" "${FILE_LIST_PATH}"
fi   
```

@designatednerd I suggest to update the Apollo docs if possible since it can be misleading (it took me quite a bit to figure why it wasn't working as I expected...)

**Comment by designatednerd** [comment]
Yeah i'm ages behind in updating this, thanks for the poke and for the detailed info!

**Comment by calvincestari** [comment]
Hi 👋🏻 - this issue should be resolved in 1.0. 1.0 is currently in alpha releases with a full 1.0 release in H2 2022. Please take a look at the [latest alpha](https://github.com/apollographql/apollo-ios/releases) and let us know if this is still an issue for you.
